samza-kv-rocksdb/src/main/scala/org/apache/samza/storage/kv/RocksDbKeyValueStore.scala

```diff
@@ -101,6 +102,8 @@ object RocksDbKeyValueStore extends Logging {
           }
         ))
 
+      metrics.newGauge("rocksdb.size-on-disk", () => FileUtils.sizeOfDirectory(new File(dir.getAbsolutePath)))
```

Comment:
1. Why not use RocksDB internal metric?
2. Regardless of 1, should not re-use the rocksdb internal metrics naming convention to report custom metrics.

Reply:
1. Two reasons: first, the behavior I recently reported of the metrics being broken (or at the very least, not predictably populated by all jobs running the latest release). Second, I was unable to determine whether the "total-sst-files-size" would include files nested within checkpoint dirs; calculating the size of the directory this way would recursively include that as well. If that is undesirable however, this should be discarded in favor of another approach.
2. Fair enough, I will rename to something else. Feel free to comment if you have suggestions.

Reply:
For 1, we should debug and fix that behavior if metrics aren't being emitted correctly. Would also be good to verify if total-sst-files-size includes files retained due to checkpoints, should be simple to verify locally, I can help set that up. I'd prefer to try these out before making this change.

Reply:
Ya you're right. Let's remove that from this PR then and circle back to it - I can't commit time immediately to debug and fix.

Created SAMZA-2374 to track.